lib/fiddle/struct.rb

```diff
@@ -261,21 +299,24 @@ def []=(*args)
       end
     end
 
+    def size=(new_size) # :nodoc:
+      raise ArgumentError, 'cannot change the size of a struct'
+    end
+
     def to_s() # :nodoc:
       super(@size)
     end
   end
 
-  # A C union wrapper
+  # A pointer to a C union
   class CUnionEntity < CStructEntity
     include PackInfo
 
     # Allocates a C union the +types+ provided.
     #
     # See Fiddle::Pointer.malloc for memory management issues.
     def CUnionEntity.malloc(types, func=nil)
-      addr = Fiddle.malloc(CUnionEntity.size(types))
-      CUnionEntity.new(addr, types, func)
+      super
     end
```

Comment:
Could you remove this method?

Reply:
I didn't add it - but removed as you've asked.